src/lib/hooks/useAttachFunds.ts

```diff
@@ -0,0 +1,34 @@
+import type { Coin } from "@cosmjs/stargate";
+import { useCallback } from "react";
+
+import { AttachFundsType } from "lib/types";
+import { fabricateFunds } from "lib/utils";
+
+interface AttachFundsParams {
+  attachFundOption: AttachFundsType;
+  assetsJson: string;
+  assetsSelect: Coin[];
+}
+
+export const useAttachFunds = ({
+  attachFundOption,
+  assetsJson,
+  assetsSelect,
+}: AttachFundsParams) => {
+  return useCallback(() => {
```

Comment:
this can be just a normal util fn that return funds, no hook or useCallback needed